### ISSUE TITLE
Users must accept Harness Invitation Before Being Added to a Group

### ISSUE THREAD

**Issue by mphammer** [harness-io/terraform-provider-harness#35]
# The Situation

* Through the Harness Manager we can create a harness_user and then add them to a harness_group (even if their email hasn't been verified yet)
* Through terraform the user must accept their invite to harness before we can add the user to the harness_group

# Expected Behavior

Similarly to using the Harness Manager, we should be able to add users to groups without them having accepted the invite. 

# Terraform Output Details

Plan Output:  
```
  # harness_add_user_to_group.group["user@company.com"] will be created
  + resource "harness_add_user_to_group" "group" {
      + group_id = "<group_id>"
      + id       = (known after apply)
      + user_id  = (known after apply)
    }

  # harness_user.users["user@company.com"] will be created
  + resource "harness_user" "users" {
      + email                              = "user@company.com"
      + id                                 = (known after apply)
      + is_email_verified                  = (known after apply)
      + is_imported_from_identity_provider = (known after apply)
      + is_password_expired                = (known after apply)
      + is_two_factor_auth_enabled         = (known after apply)
      + is_user_locked                     = (known after apply)
      + name                               = "user@company.com"
    }
```

Apply Output:  
```
harness_user.users["user@company.com"]: Creating...
harness_user.users["user@company.com"]: Creation complete after 1s
harness_add_user_to_group.group["user@company.com"]: Creating...

Error: Exception while fetching data (/addUserToUserGroup) : Invalid request: User does not exist

  on file.tf line 145, in resource "harness_add_user_to_group" "group":
 145: resource "harness_add_user_to_group" "group" {
```

# Further Analysis

```
$ terraform show 

# harness_user.users["user@company.com"]:
resource "harness_user" "users" {
    id                                 = "<user_id>"
    email                              = "user@company.com"
    name                               = "user@company.com"
    is_imported_from_identity_provider = false
    is_email_verified                  = false  # <- 'terraform show' shows the email isn't verified yet
    is_password_expired                = false
    is_two_factor_auth_enabled         = false
    is_user_locked                     = false
}
```

The terraform resource shows that the user's email hasn't been verified yet.  All our companies other users have had their emails verified before we executed the terraform code. After the user accepted his invite (verified his email) the terraform passed and the user was added to the group. 

Maybe the api-call `/addUserToUserGroup` fails if users have not been verified yet. 

# Why This Is An Issue

Our Operations team is blocked based how long it takes users to go into their email and accept the invite. Our Terraform code will keep failing until the user does so.

**Comment by mphammer** [comment]
@micahlmartin I just discussed this with my team. Please enable the feature flag. Having users auto-accepted would be great.

**Comment by micahlmartin** [comment]
Waiting for bug in the API to be fixed https://harness.atlassian.net/browse/PL-20776.

**Comment by releaseengineering** [comment]
When we add user from UI, it works with groups. Even there is API as well to create user with groups. Probably keeping user creation and add to group can be in a single resource to make It work.

**Comment by micahlmartin** [comment]
@releaseengineering agreed. I've been waiting for some changes to the graphql API in order to make this work. It might have been released recently so I'll check.

**Comment by ajaychoudhary-hotstar** [comment]
@micahlmartin Any ETA on this? this is blocking us on onboarding automation.

**Comment by micahlmartin** [comment]
Pushed a fix with version 0.1.8. Let me know if there are any problems.
